### PR TITLE
ENH: add function minimum_clearance_line

### PR DESCRIPTION
**Pull request by theroggy**
ref #1570

### CODE REVIEW

File: shapely/constructive.py [hunk 1]
```diff
@@ -595,6 +596,35 @@ def make_valid(geometry, method="linework", keep_collapsed=True, **kwargs):
         raise ValueError(f"Unknown method: {method}")
 
 
+@multithreading_enabled
+def minimum_clearance_line(geometry, **kwargs):
+    """Return a LineString whose endpoints define the minimum clearance.
+
```
Comment (jorisvandenbossche) [inline shapely/constructive.py:602]: Could you add an extended description paragraph explaining a bit more what "minimum clearance" exactly is? 
In the GEOS code itself I don't find much more than what you wrote above, but something about distance one of the vertices can be moved to become invalid.

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks @theroggy!

For the API, I was checking PostGIS and they seem to expose two versions: one that returns just the distance ([`ST_MinimumClearance`](https://postgis.net/docs/ST_MinimumClearance.html), which might often be what one wants), and one that returns the line ([`ST_MinimumClearanceLine`](https://postgis.net/docs/ST_MinimumClearanceLine.html)).

So we need to decide whether we want to do that here as well or not.

**Comment by theroggy** [comment]
> Thanks @theroggy!
> 
> For the API, I was checking PostGIS and they seem to expose two versions: one that returns just the distance ([`ST_MinimumClearance`](https://postgis.net/docs/ST_MinimumClearance.html), which might often be what one wants), and one that returns the line ([`ST_MinimumClearanceLine`](https://postgis.net/docs/ST_MinimumClearanceLine.html)).
> 
> So we need to decide whether we want to do that here as well or not.

The version that just returns the distance is already available in shapely: https://shapely.readthedocs.io/en/stable/reference/shapely.minimum_clearance.html#shapely.minimum_clearance

**Comment by jorisvandenbossche** [comment]
Whoops, sorry ;) Was going through it a bit too quickly, missed that the function being added is already called `.._line` and that we already have the basic form!

File: shapely/constructive.py (hunk 1 above)
Comment (jorisvandenbossche) [reply shapely/constructive.py:602]:
```suggestion

    A geometry's "minimum clearance" is the smallest distance by which
    a vertex of the geometry could be moved to produce an invalid geometry.

```

File: shapely/constructive.py [hunk 2]
```diff
@@ -595,6 +596,35 @@ def make_valid(geometry, method="linework", keep_collapsed=True, **kwargs):
...
+    If the geometry has no minimum clearance, an empty LineString will be returned.
+
```
Comment (jorisvandenbossche) [inline shapely/constructive.py:604]:
```suggestion

   .. versionadded:: 2.1.0
```

**Review by jorisvandenbossche — APPROVED** [review]
Thanks!

**Pull request merged**